### ISSUE TITLE
"accuracy_thresh_expand" stays unchanged 

### ISSUE THREAD

**Issue by miaoqiz** [jantic/DeOldify#122]
Hi there,

For some reason, in "Pretrain Critic" phase, the "accuracy_thresh_expand" never changed although "train_loss" and "valid_loss" went up and down a little. 

I have multiple epochs running. Even then, the "accuracy_thresh_expand" stays exactly the same. I have also specified "detach" as "False" in "hook_outputs (...)" in "unet.py". 

Is it something expected? 

Thanks!

**Comment by jantic** [comment]
accuracy_thresh_expand should definitely change.  This is what it looks like on my current run, for example:
![image](https://user-images.githubusercontent.com/179759/59232933-4a0bf480-8b9b-11e9-9ae3-cbbbc5fe60ce.png)

**Comment by miaoqiz** [comment]
Thanks for the quick reply!

For the test here, for example, in the first epoch - "Pretrain Citiic" phase:

-------------------------------------------------------------------------------------------
epoch | train_loss | valid_loss | accuracy_thresh_expand | time
-- | -- | -- | -- | --
0 | 0.316569 | 0.287891 | 0.933468 | 02:21
1 | 0.238969 | 0.245925 | 0.933468 | 02:22
2 | 0.213034 | 0.246162 | 0.933468 | 02:22
3 | 0.213239 | 0.243953 | 0.933468 | 02:23

-------------------------------------------------------------------------------------------

The dataset for testing is pretty small compared to IMAGENET. 

Both loss values are going down in general, but the "accuracy_thresh_expand" stay unchanged. 

Thanks a lot!

**Comment by jantic** [comment]
Well I guess then the question is- just how small is this dataset for testing?

**Comment by miaoqiz** [comment]
It is almost 5000 images. 

I did have a test in which "accuracy_thresh_expand" value goes up to certain value quickly and stays unchanged afterwards. 

Not sure about this but I feel that for relatively small dataset, "Generator" can boost the accuracy really fast after few epochs. It is also interesting that the temporary colorized images in "VideoModel_image_gen" folder looks pretty good, but it comes to using the same generator model to colorize the entire clip, the output is still pretty much black and white. 

Also, where is the "RGB -> YUV -> YYY" happening? the old version used to have a "transform.py", is it replaced by some call in "fastai" in the new version?

**Comment by jantic** [comment]
> Also, where is the "RGB -> YUV -> YYY" happening?
There's a function called post_process in the ColorizeFilter class in filters.py that does what you seem to be asking about.

> I did have a test in which "accuracy_thresh_expand" value goes up to certain value quickly and stays unchanged afterwards.
As far as accuracy_thresh_expand goes- I think it's very possible that it's simply not getting any more accurate and the number of images correctly classified remains unchanged.  Especially with a set that small (5000) and the fact that the value does go up for a while.  It's the simplest explanation, and it can be validated fairly easily by some manual interventions on the data.

> Not sure about this but I feel that for relatively small dataset, "Generator" can boost the accuracy really fast after few epochs.
Is this a theory of yours or is this actually happening in practice?  Sounds like the former given your results.  I've only been training on a much larger dataset (1-2 million images).  I'd suggest if you're trying to working with a small dataset that you start with the "pretrain only" weights listed in in the readme and finetune from there.

**Comment by miaoqiz** [comment]
Thanks, just wondering where the "RGB -> YUV -> YYY" happening in preparing the training dataset? perhaps you have consolidated it into "Noise Augmented Images" phase?

**Comment by jantic** [comment]
This is happening in the Jupyter notebooks (conversion from color to black/white) if that's what you're asking:  

def create_training_images(fn,i):
    dest = path_lr/fn.relative_to(path_hr)
    dest.parent.mkdir(parents=True, exist_ok=True)
    img = PIL.Image.open(fn).convert('LA').convert('RGB')
    img.save(dest)

**Comment by miaoqiz** [comment]
Sorry, I could not locate the function. Is it in "ColorizeTrainingVideo.ipynb"?

**Comment by miaoqiz** [comment]
Sorry, found it in "ColorizeTrainingArtistic.ipynb". Thanks!
